### PR TITLE
[ENH] Add filename template legend

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes https://github.com/bids-standard/bids-specification/issues/1198
replaces #1219

let's see how this one goes

### CODE REVIEW

File: tools/schemacode/bidsschematools/render.py [hunk 1]
```diff
@@ -469,11 +470,55 @@ def make_filename_template(
...
+def append_filename_template_legend(
+    text=None,
+    pdf_format=False,
+):
+
+    if text is None:
+        return
+
+    md = MarkdownIt().disable("image").enable("table")
+
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:488]: Let's move this below.

```suggestion
```

File: tools/schemacode/bidsschematools/render.py [hunk 2]
```diff
@@ -469,11 +470,55 @@ def make_filename_template(
...
+        text += f"""
+
+<details>
+<summary><strong>Legend:</strong></summary>
+<ul>
+{md.render(legend)}
+</ul>
+</details>
+
+"""
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:517]: The disable/enable things don't seem to change the code at all. Also, `<ul>` is part of the render.

```suggestion
    else:
        md = MarkdownIt()
        text += f"""
<details>
<summary><strong>Legend:</strong></summary>
{md.render(legend)}
</details>
"""
```

File: tools/schemacode/bidsschematools/render.py [hunk 3]
```diff
@@ -469,11 +470,55 @@ def make_filename_template(
...
 
 
+def append_filename_template_legend(
+    text=None,
+    pdf_format=False,
+):
+
+    if text is None:
+        return
+
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:486]: I don't see a point in allowing `text` to be `None` and having a special carveout. We take a `str`, return a `str`.

```suggestion
def append_filename_template_legend(text, pdf_format=False):
```

**Review by effigies — COMMENTED** [review]
Thank you for paring it down! It is much easier to think about when it's just the heart of it. Some small suggestions that should have no impact on the render but tighten up the function a bit.

File: tools/schemacode/bidsschematools/render.py (hunk 3 above)
Comment (Remi-Gau) [reply tools/schemacode/bidsschematools/render.py:486]: ha yeah this me wrongly assuming I was the cause of this [bug](https://github.com/bids-standard/bids-specification/pull/1219#issuecomment-1217769662) and trying things around but forgetting to delete them.

Comment (Remi-Gau) [reply tools/schemacode/bidsschematools/render.py]: FYI I a have no idea what this means:

> having a special carveout

Comment (effigies) [reply tools/schemacode/bidsschematools/render.py]: A carveout is a case that needs to be handled specifically; here we had an invalid value being checked for but also made the default.

Apologies, this is probably an Americanism, as it is often used to describe exemptions that are written into laws.

**Review by effigies — APPROVED** [review]

**Review by sappelhoff — APPROVED** [review]

File: tools/schemacode/bidsschematools/render.py (hunk 3 above)
Comment (Remi-Gau) [reply tools/schemacode/bidsschematools/render.py]: No worries
I thought it was some pythonic or well known coding lingo and google was not helpful there.

**Review by tsalo — APPROVED** [review]
💯 😄

**Comment by sappelhoff** [comment]
Thanks @Remi-Gau

**Pull request merged**